app/controllers/dispatches_controller.rb

```diff
@@ -67,9 +67,18 @@ def validate_quantity
         @project_id = dispatch_params["proj_id"]
         @quantity = dispatch_params["quantity"]
         @unit = dispatch_params["unit"]
+        @requisition_no = dispatch_params["requisition_number"]
+        @requisition = Requisition.where(:requisition_no => @requisition_no).first
+        @project_code_allocation = ProjectCodeAllocation.where(:requisition_id => @requisition.id, :project_id => @project_id).first
+        @allocated = 0
+        if ( @project_code_allocation.present? )
+            @allocated = UnitOfMeasure.find(@project_code_allocation.unit_of_measure_id).to_ref(@project_code_allocation.amount)
```

Comment:
Our refrence UOM is actually MT but what if the ref is kg ? we will be ending two different units -  adding the kg (@allocated) and the MT (@stock).